pkg/sql/sqlinstance/instancestorage/instancereader.go

```diff
@@ -141,6 +145,13 @@ func makeInstanceInfos(rows []instancerow) []sqlinstance.InstanceInfo {
 	return ret
 }
 
+// GetAllInstancesNoCache reads all instances directly from the sql_instances
+// table, bypassing any caching.
+func (r *Reader) GetAllInstancesNoCache(ctx context.Context) ([]sqlinstance.InstanceInfo, error) {
+	txn := kv.NewTxn(ctx, r.db, 0 /* gatewayNodeID */)
```

Comment:
`NewTxn` is a low-level interface for handling txn lifecycles manually. The correct API to use here is:

```go
var instances []sqlinstance.InstanceInfo
if err := r.db.Txn(ctx, func(ctx context.Context, txn *kv.Txn) (err error) {
    instances, err = r.GetAllInstancesUsingTxn(ctx, txn)
    return err
}; err != nil {
    return nil, err
}
return instances, nil
```